src/main/java/com/points/calculate/OperatorNode.java

```diff
@@ -0,0 +1,50 @@
+package com.points.calculate;
+
+import com.points.node.twochild.Node;
+
+public class OperatorNode extends Node<Operator>{
+	
+	public OperatorNode(Operator op) {
+		super(op);
+	}
+	
+	@Override
+	public Node evaluate() throws IllegalArgumentException {
```

Comment:
Why does evaluate return a Node ? It should return the result of an evaluation; which should either be Object and left up to the customer of the library to cast or the type should be specified in the generics.